solutions/platinum/cf-1254D.mdx

```diff
@@ -0,0 +1,142 @@
+---
+id: cf-1254D
+source: CF
+title: Tree Queries
+author: Rameez Parwez
+---
+
+[Official Editorial (C++)](https://codeforces.com/blog/entry/71594)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O} ((N + Q) \log(N))$
+
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int N = 150001, mod = 998244353;
+vector<int> adj[N], sum(N);
+vector<int> top(N), in(N), parent(N), depth(N), out(N), sz(N);
+int T, n, q;
+
+vector<int> bit(N);
+void add(int x, int y) {
+	while (x <= N) {
+		bit[x] = (bit[x] + y) % mod;
```

Comment:
did you test the code?

Reply:
yes

Reply:
https://codeforces.com/contest/1254/submission/269764154